zilliqa/src/contracts/deposit_v2.sol

```diff
@@ -166,44 +169,81 @@ contract Deposit {
-    mapping(address => bytes) _stakerKeys;
-
-    // The latest epoch for which the committee was calculated. It is implied that no changes have (yet) occurred in
-    // future epochs, either because those epochs haven't happened yet or because they have happened, but no deposits
-    // or withdrawals were made.
-    uint64 latestComputedEpoch;
-
-    uint256 public immutable minimumStake;
-    uint256 public immutable maximumStakers;
-
-    uint64 public immutable blocksPerEpoch;
+    /// @custom:storage-location erc7201:zilliqa.storage.DepositStorage
+    struct DepositStorage {
+        // The committee in the current epoch and the 2 epochs following it. The value for the current epoch
+        // is stored at index (currentEpoch() % 3).
+        Committee[3] _committee;
+        // All stakers. Keys into this map are stored by the `Committee`.
+        mapping(bytes => Staker) _stakersMap;
+        // Mapping from `controlAddress` to `blsPubKey` for each staker.
+        mapping(address => bytes) _stakerKeys;
+        // The latest epoch for which the committee was calculated. It is implied that no changes have (yet) occurred in
+        // future epochs, either because those epochs haven't happened yet or because they have happened, but no deposits
+        // or withdrawals were made.
+        uint64 latestComputedEpoch;
+        uint256 minimumStake;
+        uint256 maximumStakers;
+        uint64 blocksPerEpoch;
+    }
 
     modifier onlyControlAddress(bytes calldata blsPubKey) {
+        DepositStorage storage $ = _getDepositStorage();
         require(blsPubKey.length == 48);
         require(
-            _stakersMap[blsPubKey].controlAddress == msg.sender,
+            $._stakersMap[blsPubKey].controlAddress == msg.sender,
             "sender is not the control address"
         );
         _;
     }
 
-    constructor(
+    // keccak256(abi.encode(uint256(keccak256("zilliqa.storage.DepositStorage")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 private constant DepositStorageLocation =
+        0x958a6cf6390bd7165e3519675caa670ab90f0161508a9ee714d3db7edc507400;
+
+    function _getDepositStorage()
+        private
+        pure
+        returns (DepositStorage storage $)
+    {
+        assembly {
+            $.slot := DepositStorageLocation
+        }
+    }
+
+    function version() public view returns (uint64) {
+        return _getInitializedVersion();
+    }
+
+    function __Deposit_init(address initialOwner) internal onlyInitializing {
+        __Ownable2Step_init_unchained();
+        __Ownable_init_unchained(initialOwner);
+        __UUPSUpgradeable_init_unchained();
+    }
+
+    function _authorizeUpgrade(
+        address newImplementation
+    ) internal virtual override onlyOwner {}
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address initialOwner,
```

Comment:
The initializer is only needed in v1. Subsequent version have a reinitializer. We can remove the entire `initialize()` function from v2.